dash/include/dash/algorithm/Operation.h

```diff
@@ -45,19 +45,71 @@ class ReduceOperation {
   typedef ValueType value_type;
 
 public:
-  constexpr typename std::enable_if< enabled, dart_operation_t >::type
-  dart_operation() const {
+  static constexpr dart_operation_t
+  dart_operation() {
     return _op;
   }
 
-  constexpr typename std::enable_if< enabled, OpKind >::type
-  op_kind() const {
+  static constexpr OpKind
+  op_kind() {
     return _kind;
   }
 };
 
 } // namespace internal
 
+#ifdef DOXYGEN
+/**
+ * Query the underlying \ref dart_operation_t for arbitrary binary operations.
+ * Yields \c DART_OP_UNDEFINED for non-DART operations.
+ *
+ * \see      dash::min
+ * \see      dash::max
+ * \see      dash::plus
+ * \see      dash::multiply
+ * \see      dash::first
+ * \see      dash::second
+ * \see      dash::bit_and
+ * \see      dash::bit_or
+ * \see      dash::bit_xor
+ *
+ * \ingroup  DashReduceOperations
+ */
+template<typename BinaryOperation>
+struct dart_operation
+{
+  dart_operation_t value;
+}
+#else // DOXYGEN
+/**
+ * Query the DART operation for an arbitrary binary operations.
+ * Overload for non-DART operations.
+ */
+template<typename BinaryOperation, typename = void>
+struct dart_reduce_operation
+  : public std::integral_constant<dart_operation_t, DART_OP_UNDEFINED>
+{ };
+
+/**
+ * Query the DART operation for an arbitrary binary operation.
+ * Overload for DART operations.
+ */
+template<>
+template<typename BinaryOperation>
+struct dart_reduce_operation<BinaryOperation,
```

Comment:
Can you please add some comments to document this meta programming stuff. Just to understand what it does and what the `enable_if` requirements are.

Reply:
I added some documentation on the `std::enable_if` and moved it to `dash::internal`.